src/external/onerep.js

```diff
@@ -309,3 +313,48 @@ export async function listScanResults(profileId, options = {}) {
   }
   return response.json();
 }
+
+export async function isEligible() {
+  const session = await getServerSession(authOptions);
+  if (!session?.user?.subscriber?.id) {
+    throw new Error("No session");
+  }
+
+  const result = await getOnerepProfileId(session.user.subscriber.id);
+  const profileId = result[0]["onerep_profile_id"];
+  const scanResult = await getLatestOnerepScan(profileId);
+
+  if (scanResult.length) {
+    const latestScanDate = new Date(scanResult[0]["created_at"]);
+    const lastMonth = new Date();
+    lastMonth.setMonth(lastMonth.getMonth() - 1);
+
+    // FIXME only premium users get once monthly
+    if (latestScanDate > lastMonth) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ *
+ * @param {number} profileId
+ * @param {number} scanId
+ * @returns {Scan}
+ */
+export async function showScanDetails(profileId, scanId) {
```

Comment:
nit: should this be _get_ScanDetails() or _fetch_ScanDetails()?
"show" sounds possibly confusing since it's merely fetching JSON and not displaying anything.

Reply:
It's actually called both in the docs ("show" and "get"): https://docs.onerep.com/#operation/getScan

If `get` seems clearer I am fine with that.